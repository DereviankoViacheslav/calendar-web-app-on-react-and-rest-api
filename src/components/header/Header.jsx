import React from 'react';

function Header() {
  return (
    <header className="head">
      <nav className="navigate">
        <button className="navigate_create">Create</button>
        <button className="navigate_today">Today</button>
        <div className="navigate__arows">
          <div className="navigate__arows_left"></div>
          <div className="navigate__arows_right"></div>
        </div>
        <div className="navigate__MonthAndYear">Jan - Feb 2020</div>
      </nav>
    </header>
  );
}

export default Header;
