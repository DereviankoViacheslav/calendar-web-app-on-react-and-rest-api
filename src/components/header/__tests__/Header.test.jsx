import React from 'react';
import { shallow } from 'enzyme';
import Header from '../Header'
import moment from 'moment';

describe('<Header />', () => {
  const props = {
    onShowPopup: jest.fn(),
    goNextWeek: jest.fn(),
    goPrevWeek: jest.fn(),
    goToday: jest.fn(),
    date: moment('2020-02-12'),
  };

  it('Get snapshot', () => {
    const wrappedComponent = shallow(<Header {...props} />);
    expect(wrappedComponent).toMatchSnapshot();
  });

  it('should render navigate', () => {
    const wrappedComponent = shallow(<Header {...props} />);
    expect(wrappedComponent.find('.navigate').exists()).toBeTruthy();
  });
});