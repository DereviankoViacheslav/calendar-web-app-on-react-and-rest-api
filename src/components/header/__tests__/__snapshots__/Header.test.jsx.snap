// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> Get snapshot 1`] = `
<header
  className="header"
>
  <nav
    className="navigate"
  >
    <button
      className="navigate__btn-create"
      onClick={[Function]}
    >
      <svg
        height="36"
        viewBox="0 0 36 36"
        width="36"
      >
        <path
          d="M16 16v14h4V20z"
          fill="#34A853"
        />
        <path
          d="M30 16H20l-4 4h14z"
          fill="#4285F4"
        />
        <path
          d="M6 16v4h10l4-4z"
          fill="#FBBC05"
        />
        <path
          d="M20 16V6h-4v14z"
          fill="#EA4335"
        />
        <path
          d="M0 0h36v36H0z"
          fill="none"
        />
      </svg>
      Create
    </button>
    <button
      className="navigate__btn-today"
      onClick={[MockFunction]}
    >
      Today
    </button>
    <div
      className="navigate__arows"
    >
      <div
        className="navigate__arows_left"
        onClick={[MockFunction]}
      />
      <div
        className="navigate__arows_right"
        onClick={[MockFunction]}
      />
    </div>
    <div
      className="navigate__title"
    >
      Feb   2020
    </div>
  </nav>
</header>
`;
